pumps/mongo_aggregate.go

```diff
@@ -416,3 +460,25 @@ func (m *MongoAggregatePump) collectionExists(name string) (bool, error) {
 func (m *MongoAggregatePump) WriteUptimeData(data []interface{}) {
 	m.log.Warning("Mongo Aggregate should not be writing uptime data!")
 }
+
+// getLastDocumentTimestamp will return the timestamp of the last document in the collection
+func getLastDocumentTimestamp(session *mgo.Session, collectionName string) (time.Time, error) {
+	var doc bson.M
+	err := session.DB("").C(collectionName).Find(nil).Sort("-$natural").One(&doc)
+	if err != nil {
+		return time.Time{}, err
+	}
+
+	return doc["timestamp"].(time.Time), nil
```

Comment:
I think it's unsafe to return something casted like this. Please do the validation before.